digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/assetUpdater/RoadLinkPropertyUpdater.scala

```diff
@@ -180,6 +209,44 @@ class RoadLinkPropertyUpdater {
     createdProperties.flatten
   }
 
+  /**
+   * Transfers properties from old links to the new link. Each property is transfered exactly once, and if any property is missing from any old link, new link is deemed as incomplete.
+   * @param relatedChanges all changes related to the merge
+   * @param newLink
+   * @param timeStamp
+   * @return All created properties and possible incomplete link
+   */
+  def transferFunctionalClassesAndLinkTypesForSingleReplace(relatedMerges: Seq[RoadLinkChange], newLink: RoadLinkInfo, timeStamp: String): (ListBuffer[Option[ReportedChange]], ListBuffer[IncompleteLink]) = {
+    val incompleteLink = new ListBuffer[IncompleteLink]()
+    val createdProperties = new ListBuffer[Option[ReportedChange]]()
+    val transferedFunctionalClasses = new ListBuffer[FunctionalClassChange]
+    val transferedLinkTypes = new ListBuffer[LinkTypeChange]
+    relatedMerges.foreach { merge =>
```

Comment:
Voi olla, että en ymmärrä kokonaan, joten varmistan: Nyt kun tämä iteroi kaikki, niin voiko tässä tulla tilanne, jossa tulee näitä luokkia ja tyyppejä eri linkeiltä? Ja IncompleteLinkkejä tulee useampia?

Reply:
Luokkia ja tyyppejä ei operaattorin toiveesta yhdistellä eri vanhoilta linkeiltä. Transfer-yritykset lopetaan heti jos jokin vanha linkki on viallinen ja uusi linkki siirretään korjattavien listalle manuaaliseen käsittelyyn. Siten myös IncompleteLinks.size <= 0

Reply:
Siksi tuossa toteutuksessa heti ensimmäinen ehto tarkistaa että incompleLink.isEmpty

Reply:
Ok, nyt tajusin.